workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -19,7 +20,40 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
+            [Flags]
+            public enum DirtyBitsEnum
+            {
+                None = 0,
+<# int currentFieldBinaryIdx = 1; #>
+<# foreach(var fieldDetails in fieldDetailsList) { #>
+                <#= fieldDetails.PascalCaseName #> = <#= currentFieldBinaryIdx #>,
+<# currentFieldBinaryIdx = currentFieldBinaryIdx << 1; #>
+<# } #>
+            }
+
+            private DirtyBitsEnum dirtyBits;
+
+            public DirtyBitsEnum DirtyBits => dirtyBits;
+
+            private void AddDirtyBit(DirtyBitsEnum currentField)
```

Comment:
`AddDirtyBit` -> `MarkDirtyBit`
Since you arn't adding anything, just setting/marking in a bitmask

Reply:
(But I'd still prefer this one to be inlined)

Reply:
That makes sense - I've inlined them now.